catalog_tools/plots/basics.py

```diff
@@ -83,3 +85,136 @@ def plot_fmd(ax: plt.Axes, mags: np.ndarray, color: str = 'blue',
+):
+    """
+    Plots cumulative count of earthquakes in given catalog above given Mc
+    through time. Plots a line for each given Mc.
+
+    Args:
+        ax: axis where figure should be plotted
+        cat: catalog given as a pandas dataframe, should contain the column
+             "magnitude" and  either "time" or "year"
+        mcs: the list of completeness magnitudes for which we show lines
+             on the plot
+        delta_m: binning precision of the magnitudes
+        step: histogram bin size, expressed in seconds, default: 1 day
+    """
+
+    try:
+        times_list = cat["time"]
+    except KeyError:
+        raise Exception("Dataframe needs a 'time' column.")
+
+    first_time, last_time = min(times_list), max(times_list)
+    time_diff = (last_time - first_time).total_seconds()
+    bin_edges = np.arange(-1, time_diff + 1, step)
+
+    for mc in mcs:
+        cat_above_mc = cat.query(f"magnitude>={mc-delta_m/2}")
+        time_deltas = [(time - first_time) for time in cat_above_mc["time"]]
+        deltas_numeric = [delta.total_seconds() for delta in time_deltas]
+
+        counts, bin_edges = np.histogram(
+            deltas_numeric,
+            bins=bin_edges)
+        cumulative_counts = np.cumsum(counts) / np.sum(counts)
+        bin_centres = (bin_edges[:-1] + bin_edges[1:]) / 2
+        bin_centres = [first_time + dt.timedelta(seconds=centre)
+                       for centre in bin_centres]
+        ax.plot(bin_centres, cumulative_counts, label=f"Mc={np.round(mc,2)}")
+
+    ax.set_xlabel("time")
+    ax.set_ylabel("count - cumulative")
+    ax.legend()
+
+
+def plot_mags_in_time(
+    ax: plt.Axes,
+    cat: pd.DataFrame,
+    years: Optional[list] = None,
+    mcs: Optional[list] = None
+):
+    """
+    Creates a scatter plot, each dot is an event. Time shown on x-axis,
+    magnitude shown on y-axis, but also in size of the dot.
+
+    Optionally, adds lines that represent the change in completeness magnitude.
+    For example, year_bins = [2000, 2005] and mcs = [3.5, 3.0] means that
+    between 2000 and 2005, Mc is 3.5 and after 2005, Mc is 3.0.
+
+    Args:
+        ax: axis where figure should be plotted
+        cat: catalog given as a pandas dataframe, should contain the column
+             "magnitude" and  either "time" or "year"
+        years: list of years when Mc changes, sorted in increasing order
+        mcs: changed values of Mc at times given in 'years'
+    """
+
+    try:
+        cat_years = pd.to_datetime(cat["time"])
+    except KeyError:
+        try:
+            cat_years = cat["year"]
+        except KeyError:
+            raise Exception("Dataframe needs a 'year' or 'time' column.")
+
+    ax.scatter(cat_years, cat["magnitude"], cat["magnitude"]**2)
+
+    if years is not None and mcs is not None:
+        years.append(np.max(cat_years) + 1)
+        mcs.append(mcs[-1])
+        ax.step(years, mcs, where="post", c="black")
+
+    ax.set_xlabel("time")
```

Comment:
Minor change: maybe axes with capital letters? Then it would be consistent with the FMD plots